src/main.rs

```diff
@@ -86,15 +85,66 @@ fn env_version(env_var_name: &str) -> Result<Option<String>> {
 
 fn find_pants_installation() -> Result<Option<PantsConfig>> {
     if let Ok(build_root) = BuildRoot::find(None) {
-        if let Some(pants_bootstrap) = PantsBootstrap::load(&build_root)? {
-            pants_bootstrap.export_env();
-        }
         let pants_config = PantsConfig::parse(build_root)?;
         return Ok(Some(pants_config));
     }
     Ok(None)
 }
 
+#[derive(Eq, PartialEq)]
+enum ScieBoot {
+    BootstrapTools,
+    Pants,
+    PantsDebug,
+}
+
+impl ScieBoot {
+    fn env_value(&self) -> OsString {
+        match self {
+            ScieBoot::BootstrapTools => "bootstrap-tools",
+            ScieBoot::Pants => "pants",
+            ScieBoot::PantsDebug => "pants-debug",
+        }
+        .into()
+    }
+
+    fn quote<T: Into<OsString> + Debug>(value: T) -> Result<String> {
+        String::from_utf8(shell_quote::bash::escape(value))
+            .context("Shell-quoted value could not be interpreted as UTF-8.")
+    }
+
+    fn into_process(
+        self,
+        scie: String,
+        build_root: Option<PathBuf>,
+        env: Vec<(OsString, OsString)>,
+    ) -> Result<Process> {
+        Ok(match build_root.map(|br| br.join(".pants.bootstrap")) {
+            Some(pants_bootstrap) if self != Self::BootstrapTools && pants_bootstrap.is_file() => {
+                Process {
```

Comment:
This is obviously a much more robust way to do things. I read https://github.com/pantsbuild/pants/issues/16351#issuecomment-1328250334 / the associated discussion uncritically when I 1st implemented this and thought bash variables (as well as env vars) needed to be captured; thus the prior approach. Only exported env vars were making it to the pants console script from the `./pants` wrapper script though; so this is all that is needed.